plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -108,13 +108,14 @@ func (r *RuntimeClass) Validate(attributes admission.Attributes, o admission.Obj
 		return nil
 	}
 
-	pod, runtimeClass, err := r.prepareObjects(attributes)
-	if err != nil {
-		return err
-	}
-
 	if utilfeature.DefaultFeatureGate.Enabled(features.PodOverhead) {
-		err = validateOverhead(attributes, pod, runtimeClass)
+		pod, runtimeClass, err := r.prepareObjects(attributes)
```

Comment:
Once we add RuntimeClass scheduling (planned for the same release) we'll want to move this back. I'd prefer to just leave it for now, as the admission controller is only run (by default) if the feature gate is enabled.